### ISSUE TITLE
[editorial] Format doc pages using Prettier?

### ISSUE THREAD

**Issue by chalin** [open-telemetry/semantic-conventions#189]
Might maintainers consider using Prettier to format docs pages? We're happily using Prettier for the OTel website. Advantages, among others:

- Uniformity in markdown syntax layout
- We save time and energy by avoiding debates over markdown layout
- **Easier to diff changes to doc pages (because Prettier line-wraps paragraphs --- vs. a single text line per paragraph)**

The last point is one of the _main_ advantages IMHO. I was prompted to create this issue following #188.

- [x] #192

**Comment by joaopgrassi** [comment]
We should consider if it's possible to enforce prettier formatting as a PR check. Otherwise we will keep running into these issues.

**Comment by thompson-tomo** [comment]
agree and that is the intention once the rules have been applied to the docs as otherwise the pr would be quite big. this way it is split in to smaller chunks.

**Comment by thompson-tomo** [comment]
Pr to add it as a check has been done as #3015

**Comment by lmolkova** [comment]
Is there any practical need in prettier? some specific issues with inconsistent lists or italic syntax? We have yamllint and markdown lint that validate syntax. Could we configure policies that are important on markdown lint?

Adding pretties would add yet another tool and yet another complication and failed check. 

We don't have any debates over  proper markdown syntax and I have never seen any practical concern related to different list or italic syntax.

**Comment by thompson-tomo** [comment]
So the biggest benefit would be being able to run it on the generated output. As well as ensure consistent syntax for formatting which linting does not do. This would then mean that we can improve the readability of our jinja templates due to the templates being able to prioritise readability/maintainability instead of focussing on what the raw output looks like.

Prettier & linter are complementary tools.

**Comment by lmolkova** [comment]
I don't understand it. We do run linting on generated output. Our linting is not opinionated and doesn't impose arbitrary limitations like _ for italic. There is no maintainability or readability problem wrt italic choices or list styles. 

Yet another linting tool would be a source of failures. 

Notice how you can't even see the failures in CI log, you have to run it (markdownlint on the contrary shows what errors are in the logs):

prettier log https://github.com/open-telemetry/semantic-conventions/actions/runs/18942144616/job/54083463665?pr=3015

```
Checking formatting...
[warn] docs/non-normative/groups/system/cpu-metrics-guidelines.md
[warn] docs/non-normative/groups/system/design-philosophy.md
[warn] Code style issues found in 2 files. Run Prettier with --write to fix.
```

markdownlint log
```
docs\general\attributes.md:170 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
docs\general\attributes.md:171 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 3]
docs\general\attributes.md:172 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 4]
...
```

Now, prettier and markdownlint could have different opinions https://github.com/LazyVim/LazyVim/discussions/2015, https://www.joshuakgoldberg.com/blog/configuring-markdownlint-alongside-prettier/ and it's yet another task to make them agree.

---------
So we're adding more problems than we're trying to fix.

**Comment by chalin** [comment]
FYI, we use both tools for the OTel website, and both are useful and complementary. Yes, you might have to tweak some mdl config, removing some checks, because you can count on Prettier for properly format. Here are the relevant files:

- https://github.com/open-telemetry/opentelemetry.io/blob/main/.markdownlint.yaml
- https://github.com/open-telemetry/opentelemetry.io/blob/bc9473716003fd99487e16f95dd2c3a5b072bd2a/package.json#L174C1-L177C5

Prettier solves more problems than it introduces IMHO. It eliminates any debates on Markdown formatting, way beyond what mdl provides.

For any sections you don't want prettier to touch, you can always use markers `<!-- prettier-ignore-->` (for a single line) for the `-start` and `-end` variants for larger regions.

**Comment by lmolkova** [comment]
@chalin  let me rephrase my initial question - does semconv markdown style (the lack of it) causes any issues on otel.io? If it does I'd support fixing it. 

If it doesn't I'd wait until we have any practical concerns requiring opinionated style. 

We accept any valid markdown today and currently contributors don't even have to run linting tools locally - you see all issues in logs. Adding prettier will require everyone to use it locally and change spacing, list symbols and other things that don't matter.

**Comment by chalin** [comment]
> [@chalin](https://github.com/chalin) let me rephrase my initial question - does semconv markdown style (the lack of it) causes any issues on otel.io?

No it does not cause any problems.

> If it doesn't I'd wait until we have any practical concerns requiring opinionated style.

Fair enough.
 
> We accept any valid markdown today and currently contributors don't even have to run linting tools locally - you see all issues in logs. Adding prettier will require everyone to use it locally and change spacing, list symbols and other things that don't matter.

Valid arguments.

Shall we close this issue then or do you wish to discuss in the next SIG meeting?

**Comment by lmolkova** [comment]
I'd love to hear if @open-telemetry/specs-semconv-maintainers have strong preference to use prettier (or another opinionated linter) before we close it.

**Comment by joaopgrassi** [comment]
I don't have any strong preference, apart from it bringing consistency and a single style. But I don't want to hurt the contrib experience (or add more annoyances) just for the sake of it.

**Comment by thompson-tomo** [comment]
> We do run linting on generated output. 

Most of the generated content are currently excluded with the only content checked being the attribute registry pages.

> Notice how you can't even see the failures in CI log, you have to run it (markdownlint on the contrary shows what errors are in the logs):

Actually that is not the case as if you take a look at https://github.com/open-telemetry/semantic-conventions/actions/runs/18942144625/job/54083463516#step:5:1 you can see what the issues are.

In relation to:
> Adding prettier will require everyone to use it locally and change spacing, list symbols and other things that don't matter.

With the pr to add it to ci-cd it is also part of the checks command so if you are following
https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md and running `make check` nothing would change. This would ensure that prettier is a seamless part of the developer experience and no new commands to learn.

In relation to functionality not possible currently which would be addressed with prettier:
- Formatting/linting of content in front matter header blocks
- ensure empty lines between front matter header blocks and the content
- consistency for link styles
- catch trailing whitespace
- Format jinja templates for readability
